web/modules/custom/weather/src/Plugin/Block/WeatherBlock.php

```diff
@@ -80,4 +110,24 @@ public function getIp() {
     return $this->location;
   }
 
+  /**
+   * Caches data.
+   *
+   * If data is not already in cache, computed it and add to the cache.
+   * If data in cache, get it from the cache.
+   */
+  public function getCachedWeather() {
+    $cid = 'weather_cached';
+    $data = NULL;
+    if ($cache = $this->cacheBackend->get($cid)) {
+      $data = $cache->data;
+    }
+    else {
+      $data = $this->getWeather();
```

Comment:
what if the $this->getWeather() got error and did not return anything? Maybe it makes sense to return FALSE from there and handle it here?